sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/SqlDelightEnvironment.kt

```diff
@@ -88,16 +92,44 @@ class SqlDelightEnvironment(
       ModuleExtension::class.java,
     )
 
+    // We need to update the new parser definition to get the new system tables.
+    // Otherwise, the old parser is used without the updated content.
+    updateApplication {
```

Comment:
I'm surprised this is needed since the gradle task operates with classloader isolation, wouldn't we never need to "unload" the previous parser?

Reply:
At least in tests, we cache the instance too: https://github.com/AlecStrong/sql-psi/blob/cca2b4c396fb057517b7ae85292a8a65226aa6c1/core/src/main/kotlin/com/alecstrong/sql/psi/core/SqlCoreEnvironment.kt#L46
Maybe we only need to update it in tests.

Reply:
Will be removed if we merge https://github.com/AlecKazakova/sql-psi/pull/546